linkerd/duplex/src/lib.rs

```diff
@@ -24,6 +23,7 @@ struct HalfDuplex<T> {
     #[pin]
     io: T,
     direction: &'static str,
+    flushing: bool,
```

Comment:
nit/TIOLI: we can't, AFAICT, be both flushing and already shutdown, we *could* consider stuffing these into an

```rust
enum State {
   Reading,
   Flushing,
   ShuttingDown,
}
```
not sure if that matters